app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -693,7 +693,10 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
             // check if the tab history if already visible.
             if (event.eventTime - event.downTime >= ViewConfiguration.getLongPressTimeout() &&
-                TabHistoryDialogFragment().isOnTopOfBackQueue(navHost.navController)
+                navHost.navController.isOnTopOfBackQueue(
+                    TabHistoryDialogFragment::class.java.canonicalName?.substringAfterLast('.')
+                        ?: TabHistoryDialogFragment::class.java.simpleName
+                )
             ) {
```

Comment:
@MozillaNoah does this extension function look better? Instead of creating a fragment instance?